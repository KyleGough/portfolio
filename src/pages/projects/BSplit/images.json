[
  { "src": "/img/bsplit1.jpg", "caption": "Bill list" },
  { "src": "/img/bsplit2.jpg", "caption": "Homepage" },
  { "src": "/img/bsplit3.jpg", "caption": "New bill form" },
  { "src": "/img/bsplit4.jpg", "caption": "Group list" },
  { "src": "/img/bsplit5.jpg", "caption": "New group form" }
]
