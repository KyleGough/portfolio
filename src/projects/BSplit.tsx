import { useEffect } from 'react';
import Section from '../components/Section';
import Divider from '../components/Divider';
import Chip from '../components/Chip';
import Pagination from '../components/Pagination';
import ImageFigure from '../components/ImageFigure';
import Screenshots from '../components/Screenshots';

export default function BSplit() {

    useEffect(() => {
        document.title = 'BSplit: Bill Splitter Web Application';
    }, []);

    const images = [
        { src:'/img/bsplit1.jpg', caption:'Bill list' },
        { src:'/img/bsplit2.jpg', caption:'Homepage' },
        { src:'/img/bsplit3.jpg', caption:'New bill form' },
        { src:'/img/bsplit4.jpg', caption:"Group list" },
        { src:'/img/bsplit5.jpg', caption:'New group form' }
    ];

    return (
        <>
        <Section>
            <h1 className='project-title'>BSplit</h1>
            <h2 className='project-subtitle'>Web Development Technologies Coursework</h2>
            <p className='text-link-hover my-4'><time dateTime='2017-02'>February 2017</time> - <time dateTime='2017-03'>March 2017</time></p>
            <p className='mb-4'>BSplit is a web application that allows registered users to create and settle payments between friends and housemates for things such as restaurant bills, utility bills and food shopping. The app's dashboard utilises AJAX requests to prevent web page reloading to enchance user experience. Dashboard and Email notifications are sent when a new bill or group is created.</p>

            <div className='flex flex-row flex-wrap items-center mt-8'>
                <Chip name='PHP' />
                <Chip name='JavaScript' />
                <Chip name='jQuery' />
                <Chip name='SQL' />
                <Chip name='HTML' />
                <Chip name='CSS' />
                <Chip name='Bootstrap' />
            </div>
        </Section>

        <Divider />

        <ImageFigure image={images[0]} />

        <Divider />

        <Section>
            <h2 className='project-header'>Features</h2>
            <ul className='project-list'>
                <li>User registration.</li>
                <li>User authentication.</li>
                <li>Create a group of existing users by email address.</li>
                <li>Create a bill for a group.</li>
                <li>Create a bill for any set of users.</li>
                <li>Mostly AJAX content to enchance user experience.</li>
                <li>Ability to settle a payment in one or multiple payments.</li>
                <li>Displays the status of bills.</li>
                <li>Notification system for new unseen bills.</li>
                <li>Email notifications for new groups and bills.</li>
                <li>Search functionality for bills and groups.</li>
            </ul>
        </Section>

        <Divider />

        <Screenshots images={images} />

        <Divider />
        
        <Pagination
            previousTitle='To Do List'
            previousLink='/projects/todo-list'
            nextTitle='Cavern Minesweeper'
            nextLink='/projects/cavern-minesweeper'
        />
        </>
    );
}