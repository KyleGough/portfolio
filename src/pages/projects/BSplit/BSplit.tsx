import React, { useEffect } from 'react';

import { Chip } from '../../../components/Chip';
import { Divider } from '../../../components/Divider';
import { ImageFigure } from '../../../components/ImageFigure';
import { Pagination } from '../../../components/Pagination';
import { Screenshots } from '../../../components/Screenshots';
import { Section } from '../../../components/Section';
import images from './images.json';

export const BSplit: React.FC = () => {
  useEffect(() => {
    document.title = 'BSplit: Bill Splitter Web Application';
  }, []);

  return (
    <>
      <Section>
        <h1 className="project-title">BSplit</h1>
        <h2 className="project-subtitle">
          Web Development Technologies Coursework
        </h2>
        <p className="text-link-hover my-4">
          <time dateTime="2017-02">February 2017</time> -{' '}
          <time dateTime="2017-03">March 2017</time>
        </p>
        <p className="mb-4 max-w-reading">
          BSplit is a web application that allows registered users to create and
          settle payments between friends and housemates for things such as
          restaurant bills, utility bills and food shopping. The app&apos;s
          dashboard utilises AJAX requests to prevent web page reloading to
          enchance user experience. Dashboard and Email notifications are sent
          when a new bill or group is created.
        </p>

        <div className="flex flex-row flex-wrap items-center mt-8 gap-4">
          <Chip name="PHP" />
          <Chip name="JavaScript" />
          <Chip name="jQuery" />
          <Chip name="SQL" />
          <Chip name="HTML" />
          <Chip name="CSS" />
          <Chip name="Bootstrap" />
        </div>
      </Section>

      <Divider />

      <ImageFigure image={images[0]} />

      <Divider />

      <Section>
        <h2 className="project-header">Features</h2>
        <ul className="project-list">
          <li>User registration.</li>
          <li>User authentication.</li>
          <li>Create a group of existing users by email address.</li>
          <li>Create a bill for a group.</li>
          <li>Create a bill for any set of users.</li>
          <li>Mostly AJAX content to enchance user experience.</li>
          <li>Ability to settle a payment in one or multiple payments.</li>
          <li>Displays the status of bills.</li>
          <li>Notification system for new unseen bills.</li>
          <li>Email notifications for new groups and bills.</li>
          <li>Search functionality for bills and groups.</li>
        </ul>
      </Section>

      <Divider />

      <Screenshots images={images} />

      <Divider />

      <Pagination
        previousTitle="To Do List"
        previousLink="/projects/todo-list"
        nextTitle="Cavern Minesweeper"
        nextLink="/projects/cavern-minesweeper"
      />
    </>
  );
};
