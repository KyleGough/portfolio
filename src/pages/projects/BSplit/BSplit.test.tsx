import { render } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import { BSplit } from './BSplit';

describe('BSplit project page', () => {
  it('renders', async () => {
    const { container } = render(
      <BrowserRouter>
        <BSplit />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
