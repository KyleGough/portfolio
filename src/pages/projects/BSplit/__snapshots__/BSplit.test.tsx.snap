// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BSplit project page renders 1`] = `
<div>
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <h1
      class="project-title"
    >
      BSplit
    </h1>
    <h2
      class="project-subtitle"
    >
      Web Development Technologies Coursework
    </h2>
    <p
      class="text-link-hover my-4"
    >
      <time
        datetime="2017-02"
      >
        February 2017
      </time>
       
      -
       
      <time
        datetime="2017-03"
      >
        March 2017
      </time>
    </p>
    <p
      class="mb-4 max-w-reading"
    >
      BSplit is a web application that allows registered users to create and settle payments between friends and housemates for things such as restaurant bills, utility bills and food shopping. The app's dashboard utilises AJAX requests to prevent web page reloading to enchance user experience. Dashboard and email notifications are sent when a new bill or group is created.
    </p>
    <div
      class="flex flex-row flex-wrap items-center mt-8 gap-4"
    >
      <div
        class="text-chip bg-chip-light text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        PHP
      </div>
      <div
        class="text-chip bg-chip-light text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        JavaScript
      </div>
      <div
        class="text-chip bg-chip-light text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        jQuery
      </div>
      <div
        class="text-chip bg-chip-light text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        SQL
      </div>
      <div
        class="text-chip bg-chip-light text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        HTML
      </div>
      <div
        class="text-chip bg-chip-light text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        CSS
      </div>
      <div
        class="text-chip bg-chip-light text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        Bootstrap
      </div>
    </div>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <img
      alt="Bill list"
      class="border-2 border-white rounded-lg text-center items-center shadow mx-auto"
      src="/img/bsplit1.jpg"
    />
    <p
      class="mt-4 mb-2 text-sm text-center text-light font-thin"
    >
      Bill list
    </p>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <h2
      class="project-header"
    >
      Features
    </h2>
    <ul
      class="project-list"
    >
      <li>
        User registration.
      </li>
      <li>
        User authentication.
      </li>
      <li>
        Create a group of existing users by email address.
      </li>
      <li>
        Create a bill for a group.
      </li>
      <li>
        Create a bill for any set of users.
      </li>
      <li>
        Mostly AJAX content to enchance user experience.
      </li>
      <li>
        Ability to settle a payment in one or multiple payments.
      </li>
      <li>
        Displays the status of bills.
      </li>
      <li>
        Notification system for new unseen bills.
      </li>
      <li>
        Email notifications for new groups and bills.
      </li>
      <li>
        Search functionality for bills and groups.
      </li>
    </ul>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <h2
      class="text-3xl mb-12"
    >
      Screenshots
    </h2>
    <div
      class="grid grid-cols-12 gap-4"
    >
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Bill list"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/bsplit1.jpg"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Bill list
        </figcaption>
      </figure>
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Homepage"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/bsplit2.jpg"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Homepage
        </figcaption>
      </figure>
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="New bill form"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/bsplit3.jpg"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          New bill form
        </figcaption>
      </figure>
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="Group list"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/bsplit4.jpg"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          Group list
        </figcaption>
      </figure>
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="New group form"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/bsplit5.jpg"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          New group form
        </figcaption>
      </figure>
    </div>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <div
      class="flex justify-between"
    >
      <a
        class="flex items-center fill-link text-link hover:fill-link-hover hover:text-link-hover"
        href="/projects/todo-list"
      >
        <svg
          class="w-6 h-6"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
            opacity=".87"
          />
          <path
            d="M17.51 3.87L15.73 2.1 5.84 12l9.9 9.9 1.77-1.77L9.38 12l8.13-8.13z"
          />
        </svg>
        <div
          class="text-center ml-4"
        >
          <p
            class="text-sm"
          >
            Previous Project
          </p>
          <p>
            To Do List
          </p>
        </div>
      </a>
      <a
        class="flex items-center fill-link text-link hover:fill-link-hover hover:text-link-hover"
        href="/projects/cavern-minesweeper"
      >
        <div
          class="text-center mr-4"
        >
          <p
            class="text-sm"
          >
            Next Project
          </p>
          <p>
            Cavern Minesweeper
          </p>
        </div>
        <svg
          class="w-6 h-6"
          enable-background="new 0 0 24 24"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0,0h24v24H0V0z"
              fill="none"
            />
          </g>
          <g>
            <polygon
              points="6.23,20.23 8,22 18,12 8,2 6.23,3.77 14.46,12"
            />
          </g>
        </svg>
      </a>
    </div>
  </div>
</div>
`;
